src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -837,6 +839,66 @@ VideoDecoder::FrameBatchOutput VideoDecoder::getFramesPlayedInRange(
   return frameBatchOutput;
 }
 
+torch::Tensor VideoDecoder::getFramesPlayedInRangeAudio(
+    double startSeconds,
+    std::optional<double> _stopSeconds) {
+  auto stopSeconds = _stopSeconds.value_or(std::numeric_limits<double>::max());
+
+  TORCH_CHECK(
+      startSeconds <= stopSeconds,
+      "Start seconds (" + std::to_string(startSeconds) +
+          ") must be less than or equal to stop seconds (" +
+          std::to_string(stopSeconds) + ".");
+
+  validateActiveStream(AVMEDIA_TYPE_AUDIO);
+
+  StreamInfo& streamInfo = streamInfos_[activeStreamIndex_];
+
+  auto lastDecodedFrameIsPlayedAt = [&streamInfo](double seconds) {
+    auto pts = secondsToClosestPts(seconds, streamInfo.timeBase);
+    return (
+        streamInfo.lastDecodedAvFramePts <= pts and
+        pts <= streamInfo.lastDecodedAvFramePts +
+                streamInfo.lastDecodedAvFrameDuration);
+  };
+
+  // TODO-AUDIO This essentially enforce that we don't need to seek (backwards).
+  // We should remove it and seek back to the stream's beginning when needed.
+  // See test_multiple_calls
```

Comment:
It'll be easier to implement the seek logic once https://github.com/pytorch/torchcodec/pull/543 is merged